terraform/modules/happy-stack-eks/main.tf

```diff
@@ -5,36 +5,49 @@ data "kubernetes_secret" "integration_secret" {
   }
 }
 
+resource "validation_error" "mix_of_internal_and_external_services" {
+  condition = length(local.external_services) > 0 && length(local.internal_services) > 0 && var.routing_method == "DOMAIN"
+  summary   = "Invalid mix of INTERNAL and EXTERNAL services"
+  details   = "With DOMAIN routing, a mix of EXTERNAL and INTERNAL services is not permitted; only EXTERNAL and PRIVATE can be mixed"
+}
+
 locals {
   # kubernetes_secret resource is always marked sensitive, which makes things a little difficult
   # when decoding pieces of the integration secret later. Mark the whole thing as nonsensitive and only
   # output particular fields as sensitive in this modules outputs (for instance, the RDS password)
   secret       = jsondecode(nonsensitive(data.kubernetes_secret.integration_secret.data.integration_secret))
   external_dns = local.secret["external_zone_name"]
-  internal_dns = local.secret["internal_zone_name"]
 
   service_definitions = { for k, v in var.services : k => merge(v, {
-    host_match   = v.service_type == "INTERNAL" ? try(join(".", ["${var.stack_name}-${k}", "internal", local.external_dns]), "") : try(join(".", ["${var.stack_name}-${k}", local.external_dns]), "")
-    service_name = "${var.stack_name}-${k}"
+    external_host_match = var.routing_method == "CONTEXT" ? try(join(".", [var.stack_name, local.external_dns]), "") : try(join(".", ["${var.stack_name}-${k}", local.external_dns]), "")
+    host_match          = var.routing_method == "CONTEXT" ? (v.service_type == "INTERNAL" ? try(join(".", [var.stack_name, "internal", local.external_dns]), "") : try(join(".", [var.stack_name, local.external_dns]), "")) : (v.service_type == "INTERNAL" ? try(join(".", ["${var.stack_name}-${k}", "internal", local.external_dns]), "") : try(join(".", ["${var.stack_name}-${k}", local.external_dns]), ""))
```

Comment:
this is pretty difficult to read, would you mind either breaking it up or doing a multi-line conditional?